File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int max_water = -1, l = 0, r = height.size() - 1;
        while (l < r) {
            max_water = max(max_water, 
                            ((r - l) * min(height[l], height[r])));
            height[l] < height[r] ? l++ : r--;
        }
        return max_water;
    }
};

File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int maxWater = 0;
        int Vcurr = 0;
        int lo = 0, hi = height.length - 1;
        while (lo < hi) {
            Vcurr = Math.min(height[lo], height[hi]);
            maxWater = Math.max(maxWater, Vcurr * (hi - lo));
            if (height[hi] >= height[lo])
                lo++;
            else
                hi--;
        }
        return maxWater;
    }
}